src/contextify.cc

```diff
@@ -38,83 +38,85 @@ class ContextifyContext : public ObjectWrap {
     void Wrap(Handle<Object> handle);
 
     static void Init(Handle<Object> target) {
-        NanScope();
+        Nan::HandleScope scope;
 
-        Local<FunctionTemplate> ljsTmpl = NanNew<FunctionTemplate>(New);
+        Local<String> className = Nan::New("ContextifyContext").ToLocalChecked();
+
+        Local<FunctionTemplate> ljsTmpl = Nan::New<FunctionTemplate>(New);
         ljsTmpl->InstanceTemplate()->SetInternalFieldCount(1);
-        ljsTmpl->SetClassName(NanNew("ContextifyContext"));
-        NODE_SET_PROTOTYPE_METHOD(ljsTmpl, "run",       ContextifyContext::Run);
-        NODE_SET_PROTOTYPE_METHOD(ljsTmpl, "getGlobal", ContextifyContext::GetGlobal);
+        ljsTmpl->SetClassName(className);
+        Nan::SetPrototypeMethod(ljsTmpl, "run",       ContextifyContext::Run);
+        Nan::SetPrototypeMethod(ljsTmpl, "getGlobal", ContextifyContext::GetGlobal);
 
-        NanAssignPersistent(jsTmpl, ljsTmpl);
-        target->Set(NanNew("ContextifyContext"), ljsTmpl->GetFunction());
+        jsTmpl.Reset(ljsTmpl);
+        target->Set(className, ljsTmpl->GetFunction());
     }
 
     static NAN_METHOD(New) {
-        NanScope();
-
-        if (args.Length() < 1) {
-            NanThrowError("Wrong number of arguments passed to ContextifyContext constructor");
-            NanReturnUndefined();
+        if (info.Length() < 1) {
+            Nan::ThrowError("Wrong number of arguments passed to ContextifyContext constructor");
+            info.GetReturnValue().SetUndefined();
```

Comment:
Thanks for commenting! Actually, after reading a bit more about V8, it seems that it is wrong, since you're technically not allowed to perform any more JS-land operations after throwing, and must bail out to engine code immediately. Any thoughts on this?